client/src/rpc_filter.rs

```diff
@@ -16,21 +16,28 @@ impl RpcFilterType {
                 match encoding {
                     MemcmpEncoding::Binary => {
                         let MemcmpEncodedBytes::Binary(bytes) = &compare.bytes;
-                        bs58::decode(&bytes)
-                            .into_vec()
-                            .map(|_| ())
-                            .map_err(|e| e.into())
+
+                        if bytes.len() > 128 {
+                            Err(RpcFilterError::Base58DataTooLarge)
```

Comment:
A quick survey of recent RPC requests for api.mainnet-beta.solana.com seemed to indicate that this new limit wouldn't impose a problem. 

But if we receive feedback that 128 is not enough we can always add a base64 encoding.  I didn't do this for now because I'm not sure we want to support such large memcpy filters regardless so I'm eager to first have the discussion about why 128 is not enough

Reply:
I'd started on the latter route before nerd sniping myself into removing that pesky redundant `encoding` field from the `Memcmp` struct :upside_down_face: